app/models/student.rb

```diff
@@ -54,6 +54,11 @@ class Student < Role
     Settings.student_csv_order || %w[user_name section_name last_name first_name id_number email]
   ).map(&:to_sym).freeze
 
+  def can_cancel_test?(test_run_id)
+    test_run = test_runs.in_progress.where(id: test_run_id).first
```

Comment:
Rather than doing a query and checking `.nil?`, look up the `exists?` ActiveRecord method